lightning/impl/sag.py

```diff
@@ -1,8 +1,20 @@
 # Author: Mathieu Blondel
+#         Arnaud Rachez
+#         Fabian Pedregosa
 # License: BSD
 
 import numpy as np
 
+try:
+    from sklearn.linear_model.sag_fast import get_max_squared_sum
+except ImportError:
+    # compatibility for older scikit-learn
+    # this is orders of magnitude slower than the one in scikit-learn,
+    # but does not need to convert X to a dense matrix
+    def get_max_squared_sum(X):
+        n = X.shape[0]
+        return np.max(X.dot(X.T)[range(n), range(n)])
+
```

Comment:
I would use `np.sum(X ** 2, axis=1).max()` in the dense case and `csr_row_norms(X).max()` in the CSR case (from sklearn.utils.sparsefuncs).